File: lib/rin/gui/events/KeyEvent.cpp
#include "rin/gui/events/KeyEvent.h"
namespace rin::gui
{
    KeyEvent::KeyEvent(Surface* inSurface, const io::CursorButton& inButton, const io::InputState& inState, const Flags<io::InputModifier>& inModifiers)  : Event(inSurface), button(inButton), state(inState), modifiers(inModifiers)
    {
        
    }
}

File: rin.Framework/Graphics/Windows/Events/KeyEvent.cs
namespace rin.Framework.Graphics.Windows.Events;

public class KeyEvent : WindowEvent
{
    public required InputKey Key;
    public required InputModifier Modifiers;
    public required InputState State;
    public bool IsAltDown => Modifiers.HasFlag(InputModifier.Alt);
    public bool IsControlDown => Modifiers.HasFlag(InputModifier.Control);
    public bool IsShiftDown => Modifiers.HasFlag(InputModifier.Shift);
}